src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -55,12 +60,28 @@ public synchronized void clearCurrentCommand() {
 
 	@Override
 	public synchronized String getCurrentCommandString() {
-		if (getCurrentCommand() == null) {
-			return "No command loaded";
-		} else
-			return getCurrentCommand().toString();
+		if (getCurrentCommand() == null ) {
+			return "No command loaded, press 'run command' to show statistics";
+		} else{
+			currentCommand = getCurrentCommand();
+
+			VisitorCounter commandVisitorCounter = new VisitorCounter();
+			CommandsFeature.getDriverCommandManager().getCurrentCommand().accept((VisitorCommand) commandVisitorCounter);
+
+			String command = getCurrentCommand().toString();
+			int operateToCounter = commandVisitorCounter.getCounter();
+			int setPositionCounter = commandVisitorCounter.getCounter();
+			int numberOfSubcommands = operateToCounter + setPositionCounter;
+
+			return("Current command: " + command + "\n"
+					+ "Number of Subcommands: " + numberOfSubcommands + "\n"
+					+ "Number of operateTo command: " + operateToCounter + "\n"
+					+ "Number of setPosition command: " + setPositionCounter + "\n"
+			);
+		}
```

Comment:
Too much modification here. I suggest printing command stats inside a log panel (using logger) or inside CommandWindow. A user could make a request by running a menu option or you can use an observer approach (better): each time new current command is set observer would print stats. Observer can be fixed or enable/disabled using menu option (better).

Reply:
Could not remove the modification and move to CommandManagerWindow, because the statistic data are not able to be updated.
Some of statistics are already displayed in logger, it was implemented already by other group.